### DISCUSSION TITLE
missing enums with fields

### DISCUSSION THREAD

**Discussion by pznknmtb** [vlang/v#11584] [Ideas]
some languages support enums with extra fields.

for according java code see `public enum Planet` on https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

as example  `tokens.v` could be more clean and efficient (omitting code like `fn build_token_str()`)

```
enum Token {
        INT('int')
        STRING('string')
        ...
```

**Comment by medvednikov** [comment]
Yes, this will be implemented.
